### PR TITLE
Separated period configuration for system/package dataset updates

### LINKED ISSUE #15877: [Auditbeat] system/package dataset updates too often
**Describe the enhancement:**

The `system/package` dataset fetches the list of installed packages every 10s by default (`system.period`), in order to check for newly installed packages. In some cases, for example rpm systems with a lot of installed packages, this frequent check causes a lot of unnecessary CPU usage by Auditbeat.

The problem with `system.period` is that most system datasets rely on this period to poll for changes, so setting it to a higher value can negatively impact other datasets. Adding a dedicated setting, like `system.package.poll.period`, similar to `system.package.state.period`, will be a better solution imho.

Alternatively, the time it takes to fetch the list of packages could be measured and the poll interval adjusted within some bounds (10s ... 5min) to ensure that only a minimum percentage of time is spent checking for new packages.

As an example, right now Auditbeat peaks to 50% CPU for ~1s every 10s, in an rpm system with ~800 packages installed, which some users find unnacceptable.

### PR DESCRIPTION
**Pull request by rvillablanca**
## What does this PR do?

Add a separated configuration for dataset that scan for newly installed packages in module system.

## Why is it important?

Because currently the `system/package` dataset fetches every 10 seconds which causes unnecessary cpu usage when scanning a large number of newly installed packages. The 10 seconds correspond to `system.period` so changing this value is not an option since it is used by others datasets.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

## Related issues
- Closes #15877

### CODE REVIEW

**Comment by rvillablanca** [comment]
@adriansr I have to define the property `package.poll.period` in some yaml configuration file but I understand that `auditbeat.reference.yml` is generated automatically so I'm not sure where I should place it. I was looking for `config.yml` in the `_meta` folder but there is no such file.

**Comment by rvillablanca** [comment]
I just found the config template `config.yml.tmpl` so I added the new config option there

**Comment by adriansr** [comment]
jenkins, test this

**Comment by adriansr** [comment]
@rvillablanca the `make check` step is failing with:
```
Error: some files are not up-to-date. Run 'mage fmt update' then review and commit the changes. Modified: [x-pack/auditbeat/docs/modules/system.asciidoc]
```

You need to run `mage update` inside `x-pack/auditbeat` and/or `make update` at top level, then commit the result, as some parts of docs are autogenerated from a template you modified. Then run `make check` at top level to ensure there are no errors.

File: x-pack/auditbeat/auditbeat.yml [hunk 1]
```diff
@@ -47,11 +47,16 @@ auditbeat.modules:
   - /usr/sbin
   - /etc
 
+- module: system
+  datasets:
+    - package # Installed, updated, and removed packages
+
+  period: 30m # The frequency at which the datasets check for changes
```
Comment (adriansr) [inline x-pack/auditbeat/auditbeat.yml]: I think we should find a compromise between the original 10s and 30m. We don't want to overload the system but at the same time we need to be responsive to catch a package that is installed for a short time. What about 2-3 minutes? @andrewkroh what do you think?

Another improvement to consider in the future would be monitor some arch-specific file for changes and then trigger the enumeration of packages.

**Comment by rvillablanca** [comment]
> @rvillablanca the `make check` step is failing with:
> 
> ```
> Error: some files are not up-to-date. Run 'mage fmt update' then review and commit the changes. Modified: [x-pack/auditbeat/docs/modules/system.asciidoc]
> ```
> 
> You need to run `mage update` inside `x-pack/auditbeat` and/or `make update` at top level, then commit the result, as some parts of docs are autogenerated from a template you modified. Then run `make check` at top level to ensure there are no errors.

It is strange because I ran the command `mage fmt update` inside `x-pack/auditbeat` and the files were update successfully, even I ran `mage check` and it is ok also. I will do the same in the at top level.

**Comment by rvillablanca** [comment]
Which is the command to update the docs ? :thinking:

**Comment by rvillablanca** [comment]
I had to run the command `mage fmt update` inside `./auditbeat` in order to regenerate the file `x-pack/auditbeat/docs/modules/system.asciidoc`. Not sure if that is ok or could be a bug, because it makes sense to me that by running `mage fmt update` inside of any beat of x-pack sould update its own doc files.

**Comment by rvillablanca** [comment]
Now CI is passing :smiley:

File: x-pack/auditbeat/auditbeat.yml (hunk 1 above)
Comment (rvillablanca) [reply x-pack/auditbeat/auditbeat.yml]: I'll be waiting for the final default value

Comment (andrewkroh) [reply x-pack/auditbeat/auditbeat.yml]: `2m` sounds good to me.

Comment (rvillablanca) [reply x-pack/auditbeat/auditbeat.yml]: :+1:

**Comment by adriansr** [comment]
jenkins, test this!

**Comment by rvillablanca** [comment]
So this is ready to be merged

**Pull request merged**